src/background/adblocker.js

```diff
@@ -214,7 +214,14 @@ async function injectCosmetics(msg, sender) {
   }
 
   // Extract hostname from sender's URL
-  const { url = '', frameId } = sender;
+  const { frameId } = sender;
+  let url = sender.url;
+
+  // Try `sender.tab.url` as a fallback when Safari preloads a website
+  if (!url || url === 'about:blank') {
+    url = sender.tab.url;
```

Comment:
Can't we just always use `sender.tab.url`?

Reply:
MDN docs specifies that `sender.tab` can be `undefined`, I don't know if there's a case without `sender.tab`.